kornia/geometry/camera/stereo.py

```diff
@@ -0,0 +1,287 @@
+    def Q(self) -> torch.Tensor:
+        """
+        The Q matrix of the horizontal stereo setup. This matrix is used for reprojecting a disparity tensor to
+        the corresponding point cloud. Note that this is in a general form that allows different focal
+        lengths in the x and y direction.
+
+        Return:
+            torch.Tensor: The Q matrix of shape :math:`(B, 4, 4)`.
+        """
+        Q = torch.zeros((self.batch_size, 4, 4), device=self.device, dtype=self.dtype)
+        baseline = -self.tx
+        Q[:, 0, 0] = self.fy * baseline
+        Q[:, 0, 3] = -self.fy * self.cx_left * baseline
+        Q[:, 1, 1] = self.fx * baseline
+        Q[:, 1, 3] = -self.fx * self.cy * baseline
+        Q[:, 2, 3] = self.fx * self.fy * baseline
+        Q[:, 3, 2] = -self.fy
+        Q[:, 3, 3] = self.fy * (self.cx_left - self.cx_right)  # NOTE: This is usually zero.
+        return Q
+
+    def reproject_disparity_to_3D(self, disparity_tensor: torch.Tensor) -> torch.Tensor:
+        """
+        Reproject the disparity tensor to a 3D point cloud.
+
+        Args:
+            disparity_tensor (torch.Tensor): Disparity tensor of shape :math:`(B, H, W)`.
+
+        Returns:
+            torch.Tensor: The 3D point cloud of shape :math:`(B, H * W, 3)`
+        """
+        return reproject_disparity_to_3D(disparity_tensor, self.Q)
+
+
+def _check_disparity_tensor(disparity_tensor: torch.Tensor):
+    """
+    Utility function to ensure correct user provided correct disparity tensor.
+    Args:
+        disparity_tensor (torch.Tensor): The disparity tensor of shape :math:`(B, H, W)`.
+    """
+    if disparity_tensor.ndim != 3:
+        raise StereoException(f"Expected 'disparity_tensor' to have 3 dimensions." f"Got {disparity_tensor.ndim}.")
+
+    if disparity_tensor.dtype != torch.float32:
+        raise StereoException(
+            f"Expected 'disparity_tensor' to have dtype torch.float32." f"Got {disparity_tensor.dtype}"
+        )
+
+
+def _check_Q_matrix(Q_matrix: torch.Tensor):
+    """
+    Utility function to ensure Q matrix is of correct form.
+    Args:
+        Q_matrix (torch.Tensor): The Q matrix for reprojecting disparity to a point cloud of shape :math:`(B, 4, 4)`
+    """
+    if not Q_matrix.ndim == 3:
+        raise StereoException(f"Expected 'Q_matrix to have 3 dimenstions." f"Got {Q_matrix.ndim}")
+
+    if not Q_matrix.shape[1:] == (4, 4):
+        raise StereoException(
+            f"Expected last two dimensions of 'Q_matrix' to be of shape (4, 4)." f"Got {Q_matrix.shape}"
+        )
+
+    if not Q_matrix.dtype == torch.float32:
+        raise StereoException(f"Expected 'Q_matrix' to be of type torch.float32." f"Got {Q_matrix.dtype}")
+
+
+def reproject_disparity_to_3D(disparity_tensor: torch.Tensor, Q_matrix: torch.Tensor) -> torch.Tensor:
+    """
+    Reproject the disparity tensor to a 3D point cloud.
+
+    Args:
+        disparity_tensor (torch.Tensor): Disparity tensor of shape :math:`(B, H, W)`.
```

Comment:
```suggestion
        disparity_tensor (torch.Tensor): Disparity tensor of shape :math:`(B, 1, H, W)`.
```
we try to represent like this single channel images, but open to discussion.